### PR TITLE
Feature/lvpytttn 58: icons instead of selects in spending registration

### PR DESCRIPTION
**Pull request by Rocckk**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=896)

## Code reviewers

- [ ] @antooa
- [ ] @HalynaDom
- [ ] @OlexaNdrus
- [ ] @pinkfoxxx
- [ ] @Rocckk
- [ ] @Rostyk1404
- [ ] @Sent1nelDNoir
- [ ] @Svviat
- [ ] @ValentynLiashenko

### Second Level Review

- [ ] @Blackleaves
- [ ] @yevgenbehen
- [ ] @lhalam

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

### CODE REVIEW

File: iBudget/fund/views.py [hunk 1]
```diff
@@ -30,11 +31,15 @@ def show_fund(request):
             HttpResponse object.
     """
     user = request.user
+    icon_if_none = \
+        'https://s3.amazonaws.com/family-finance-tracker-static/standard_fund/funds.png'
```
Comment (yevgenbehen) [inline iBudget/fund/views.py]: This is a bad idea to have such hardcoded URLs in code. Ideally it should be built with parts that are constants. For instance, `icon_if_none = AWS_S3_URL + STANDARD_FUNDS_LOCATION + icon_file_name` where `AWS_S3_URL = https://s3.amazonaws.com`, `STANDARD_FUNDS_LOCATION = /family-finance-tracker-static/standard_fund/` and `icon_file_name=funds.png`.

File: iBudget/fund/views.py [hunk 2]
```diff
@@ -50,13 +55,18 @@ def show_fund_by_group(request):
 
     user = request.user
     users_group = []
+    icon_if_none = \
+        'https://s3.amazonaws.com/family-finance-tracker-static/standard_fund/funds.png'
```
Comment (yevgenbehen) [inline iBudget/fund/views.py]: And this one is a duplication of hardcoded string that is even worse :)

File: iBudget/group/views.py [hunk 3]
```diff
@@ -33,10 +33,14 @@ def get_by_group(request):
     """
 
     user = request.user
+    icon_if_none = \
+        'https://s3.amazonaws.com/family-finance-tracker-static/standard_group/group.png'
```
Comment (yevgenbehen) [inline iBudget/group/views.py]: one more

File: iBudget/fund/views.py [hunk 4]
```diff
@@ -30,11 +31,15 @@ def show_fund(request):
...
     user = request.user
+    icon_if_none = \
+        'https://s3.amazonaws.com/family-finance-tracker-static/standard_fund/funds.png'
     if user:
         user_funds = []
         for entry in FundCategories.filter_by_user(user):
             if not FinancialGoal.has_goals(fund_id=entry.id):
-                user_funds.append({'id': entry.id, 'name': entry.name})
+                user_funds.append({'id': entry.id, 'name': entry.name,
+                                   'url': AwsService.get_image_url(entry.icon) if entry.icon
```
Comment (yevgenbehen) [inline iBudget/fund/views.py]: `url` should be built in a local variable. It will increase readability fo this code.

File: iBudget/fund/views.py [hunk 5]
```diff
@@ -50,13 +55,18 @@ def show_fund_by_group(request):
...
     if user:
         for group in Group.filter_groups_by_user_id(user):
             for shared_fund in SharedFunds.objects.filter(group=group.id):
                 if not FinancialGoal.has_goals(fund_id=shared_fund.fund.id):
+                    icon = FundCategories.objects.get(id=shared_fund.fund.id).icon
                     users_group.append({'id_fund': shared_fund.fund.id,
                                         'name_fund': shared_fund.fund.name,
-                                        'id_group': group.id
+                                        'id_group': group.id,
+                                        'url': AwsService.get_image_url(icon) if icon else
```
Comment (yevgenbehen) [inline iBudget/fund/views.py]: The same here

File: iBudget/group/views.py [hunk 6]
```diff
@@ -33,10 +33,14 @@ def get_by_group(request):
...
     if user:
         user_groups = []
         for entry in Group.filter_groups_by_user_id(user):
-            user_groups.append({'id': entry.id, 'name': entry.name})
+            user_groups.append({'id': entry.id, 'name': entry.name,
+                                'url': AwsService.get_image_url(entry.icon) if entry.icon else
```
Comment (yevgenbehen) [inline iBudget/group/views.py]: The same here

File: iBudget/spending/views.py [hunk 7]
```diff
@@ -25,10 +26,14 @@ def show_spending_ind(request):
             HttpResponse object.
     """
     user = request.user
+    icon_if_none =\
+        'https://family-finance-tracker-static.s3.amazonaws.com/standard/miscellaneous.png'
```
Comment (yevgenbehen) [inline iBudget/spending/views.py]: The same here

File: iBudget/spending/views.py [hunk 8]
```diff
@@ -25,10 +26,14 @@ def show_spending_ind(request):
...
     if user:
         user_categories = []
         for entry in SpendingCategories.filter_by_user(user):
-            user_categories.append({'id': entry.id, 'name': entry.name})
+            user_categories.append({'id': entry.id, 'name': entry.name,
+                                    'url': AwsService.get_image_url(entry.icon) if entry.icon else
```
Comment (yevgenbehen) [inline iBudget/spending/views.py]: The same here

File: iBudget/spending/views.py [hunk 9]
```diff
@@ -45,13 +50,18 @@ def show_spending_group(request):
 
     user = request.user
     users_group = []
+    icon_if_none = \
+        'https://family-finance-tracker-static.s3.amazonaws.com/standard/miscellaneous.png'
```
Comment (yevgenbehen) [inline iBudget/spending/views.py]: The same here

File: iBudget/spending/views.py [hunk 10]
```diff
@@ -45,13 +50,18 @@ def show_spending_group(request):
...
         for group in Group.filter_groups_by_user_id(user):
             for shared_category in SharedSpendingCategories.objects.filter(group=group.id):
+                icon =\
+                    SpendingCategories.objects.get(id=shared_category.spending_categories.id).icon
                 users_group.append({'id_cat': shared_category.spending_categories.id,
                                     'name_cat': shared_category.spending_categories.name,
                                     'id_group': group.id,
-                                    'name_group': group.name
+                                    'name_group': group.name,
+                                    'url': AwsService.get_image_url(icon) if icon else icon_if_none
```
Comment (yevgenbehen) [inline iBudget/spending/views.py]: The same here

File: iBudget/fund/views.py (hunk 1 above)
Comment (Rocckk) [reply iBudget/fund/views.py]: thanks, I will rewrite it like you said.

File: iBudget/fund/views.py (hunk 2 above)
Comment (Rocckk) [reply iBudget/fund/views.py]: no problem, I will correct!

File: iBudget/group/views.py (hunk 3 above)
Comment (Rocckk) [reply iBudget/group/views.py]: ++

File: iBudget/fund/views.py (hunk 4 above)
Comment (Rocckk) [reply iBudget/fund/views.py]: will be corrected!

**Pull request merged**